### PR TITLE
SublimeBidAdapter: Update to version 0.5.1

### PR DESCRIPTION
**Pull request by SublimeLeo**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

Hello, we made changes on our adapter!
We modified the code style, added some pixels to have more visibility on our adapter behaviour and made debug easier.

- test parameters for validating bids
```
{
  bidder: 'sublime',
  params: {
    zoneId: 1
  }
}
```

- contact email of the adapter’s maintainer: pbjs@sublime.xyz

### CODE REVIEW

**Comment by robertrmartinez** [comment]
Hello, taking a look at this now.

We do not allow adapters to fire pixels inside of non registered pixel events, so sending an event for every bidResponse is not allowed.

Adapters should only be sending pixels on the allowed events like,

onTimeout, onBidWon etc

File: modules/sublimeBidAdapter.js [hunk 1]
```diff
@@ -1,136 +1,217 @@
...
+ *
+ * @param {*} serverResponse A successful response from the server.
+ * @param {*} bidRequest An object with bid request informations
+ * @return {Bid[]} An array of bids which were nested inside the server.
+ */
+function interpretResponse(serverResponse, bidRequest) {
+  const bidResponses = [];
+  const response = serverResponse.body;
+
+  sendEvent('dintres');
```
Comment (robertrmartinez) [inline modules/sublimeBidAdapter.js:154]: pixel fires outside of the pixel listen hooks is not allowed. please remove this.

File: modules/sublimeBidAdapter.js [hunk 2]
```diff
@@ -1,136 +1,217 @@
...
+      currency: response.currency || DEFAULT_CURRENCY,
+      netRevenue: response.netRevenue || true,
+      ttl: response.ttl || DEFAULT_TTL,
+      ad: response.ad,
+      pbav: SUBLIME_VERSION
+    };
 
-      bidResponses.push(bidResponse);
-    }
+    sendEvent('bida', true);
```
Comment (robertrmartinez) [inline modules/sublimeBidAdapter.js:190]: same as above, please remove

File: modules/sublimeBidAdapter.js [hunk 3]
```diff
@@ -1,136 +1,217 @@
...
+    bidResponses.push(bidResponse);
+  } else {
+    sendEvent('dnobid');
```
Comment (robertrmartinez) [inline modules/sublimeBidAdapter.js:193]: please remove

File: modules/sublimeBidAdapter.js [hunk 4]
```diff
@@ -1,136 +1,217 @@
...
 
-    return bidResponses;
-  },
+/**
+ * Send debug when we timeout
+ * @param {Object} timeoutData
+ */
+function onTimeout(timeoutData) {
+  log('Timeout from adapter', timeoutData);
+  sendEvent('dbidtimeout', true);
```
Comment (robertrmartinez) [inline modules/sublimeBidAdapter.js:205]: pixels are allowed onTimeout so you can leave this

**Comment by SublimeLeo** [comment]
Hello @robertrmartinez, sorry for the delay!
We made all the requested changes!

**Comment by SublimeLeo** [comment]
Oops, it seems that we have errors with the CI. 
I will fix it asap and I will notify you when it's done!

**Comment by SublimeLeo** [comment]
CI should be fine now!

**Comment by robertrmartinez** [comment]
Hi sorry for the delay, have been swamped.

Will get this reviewed and merged tomorrow!

**Comment by SublimeLeo** [comment]
Cool, thank you! Let me know if I missed anything!

**Pull request merged**